pkg/kates/accumulator.go

```diff
@@ -165,47 +173,61 @@ func newAccumulator(ctx context.Context, client *Client, queries ...Query) (*Acc
 		client.watchRaw(ctx, q, rawUpdateCh, client.cliFor(field.mapping, q.Namespace))
 	}
 
-	acc := &Accumulator{client, fields, map[string]bool{}, 0, changed, sync.Mutex{}}
-
-	// This coalesces reads from rawUpdateCh to notifications that changes are available to be
-	// processed. This loop along with the logic in storeField guarantees the 3
-	// Goals/Requirements listed in the documentation for the Accumulator struct, i.e. Ensuring
-	// all Kinds are bootstrapped before any notification occurs, as well as ensuring that we
-	// continue to coalesce updates in the background while business logic is executing in order
-	// to ensure graceful load shedding.
-	go func() {
-		canSend := false
-
-		for {
-			var rawUp rawUpdate
-			if canSend {
-				select {
-				case changed <- struct{}{}:
-					canSend = false
-					continue
-				case rawUp = <-rawUpdateCh:
-				case <-ctx.Done():
-					return
-				}
-			} else {
-				select {
-				case rawUp = <-rawUpdateCh:
-				case <-ctx.Done():
-					return
-				}
-			}
+	acc := &Accumulator{
+		client:   client,
+		fields:   fields,
+		excluded: map[string]bool{},
+		synced:   0,
+		changed:  changed,
+		mutex:    sync.Mutex{},
+	}
 
-			// Don't overwrite canSend if storeField returns false. We may not yet have
-			// had a chance to send a notification down the changed channel.
-			if acc.storeUpdate(rawUp) {
-				canSend = true
-			}
-		}
-	}()
+	go acc.Listen(ctx, rawUpdateCh, client.maxAccumulatorInterval)
 
 	return acc, nil
 }
 
+// This coalesces reads from rawUpdateCh to notifications that changes are available to be
+// processed. This loop along with the logic in storeField guarantees the 3
+// Goals/Requirements listed in the documentation for the Accumulator struct, i.e. Ensuring
+// all Kinds are bootstrapped before any notification occurs, as well as ensuring that we
+// continue to coalesce updates in the background while business logic is executing in order
+// to ensure graceful load shedding.
+func (a *Accumulator) Listen(ctx context.Context, rawUpdateCh chan rawUpdate, interval time.Duration) {
```

Comment:
I would narrow types arguments to indicate that this channel is only for reading `rawUpdateCh <-chan rawUpdate`